### PR TITLE
Ticket/292 - part 1

### PR DESCRIPTION
**Pull request by melinath**
Resolves half of #292 - display of whether an order has pending payments in the organizer model tables.

### CODE REVIEW

**Comment by melinath** [comment]
The other half - displaying pending payments nicely on the user page - will still need to be done.

File: brambling/models.py [hunk 1]
```diff
@@ -651,13 +664,13 @@ def remove_from_cart(self, bought_item):
             self.cart_start_time = None
             self.save()
 
-    def mark_cart_paid(self):
+    def mark_cart_paid(self, status):
```
Comment (harrislapiroff) [inline brambling/models.py]: Should this method be renamed, since it no longer marks a cart as paid, necessarily?

File: brambling/models.py [hunk 2]
```diff
@@ -775,17 +784,44 @@ class Payment(models.Model):
     card = models.ForeignKey('CreditCard', blank=True, null=True)
     is_confirmed = models.BooleanField(default=False)
 
+    def refund(self, amount, issuer, dwolla_pin=None):
```
Comment (harrislapiroff) [inline brambling/models.py]: What happens if someone refunds an order paid by check? I'm not sure I totally understand the code.

**Comment by harrislapiroff** [comment]
The code here looks good. I've put a couple questions inline. I'm gonna start testing things out now, but this is hard code to test manually. It feels like a reminder to me that we really need to get moving on automated testing and CI.

**Comment by harrislapiroff** [comment]
Checking out with a person who already had a couple passes purchased, I received this exception on the summary page:

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/harris-dance-weekend/order/summary/

Django Version: 1.7.2
Python Version: 2.7.6
Installed Applications:
('brambling',
 'grappelli',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'zenaida',
 'zenaida.contrib.hints',
 'floppyforms',
 'django_filters',
 'daguerre',
 'debug_toolbar.apps.DebugToolbarConfig',
 'template_timings_panel',
 'zenaida.contrib.feedback')
Installed Middleware:
(u'debug_toolbar.middleware.DebugToolbarMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'zenaida.contrib.feedback.middleware.FeedbackMiddleware')


Traceback:
File "/Users/harris/.virtualenvs/brambling/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  111.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/Users/harris/.virtualenvs/brambling/lib/python2.7/site-packages/django/views/generic/base.py" in view
  69.             return self.dispatch(request, *args, **kwargs)
File "/Users/harris/Projects/Clients/Little Weaver/django-brambling/brambling/views/orders.py" in dispatch
  254.         return super(OrderMixin, self).dispatch(request, *args, **kwargs)
File "/Users/harris/.virtualenvs/brambling/lib/python2.7/site-packages/django/views/generic/base.py" in dispatch
  87.         return handler(request, *args, **kwargs)
File "/Users/harris/Projects/Clients/Little Weaver/django-brambling/brambling/views/orders.py" in get
  750.             self.get_forms()
File "/Users/harris/Projects/Clients/Little Weaver/django-brambling/brambling/views/orders.py" in get_forms
  827.             self.check_form = CheckPaymentForm(data=check_data, **kwargs)
File "/Users/harris/Projects/Clients/Little Weaver/django-brambling/brambling/forms/orders.py" in __init__
  395.                              if item['net_balance'] > 0]

Exception Type: KeyError at /harris-dance-weekend/order/summary/
Exception Value: 'net_balance'
```

File: brambling/models.py (hunk 1 above)
Comment (melinath) [inline brambling/models.py]: Maybe. Suggestions?

File: brambling/models.py (hunk 2 above)
Comment (melinath) [inline brambling/models.py]: The refund is always listed as the same method as the payment. Which is a little weird and not necessarily accurate. For example, I might accept payment in cash but refund someone via paypal (hypothetically).

**Comment by melinath** [comment]
@harrislapiroff that error is corrected in pull request #304 :-p maybe you'll need to do both of them at once after all.

**Comment by melinath** [comment]
Or I guess I could do a backport.

File: brambling/models.py (hunk 1 above)
Comment (harrislapiroff) [inline brambling/models.py]: `mark_cart_status`?

Comment (melinath) [inline brambling/models.py]: Looking at this again, I think mark_cart_paid is appropriate. The "cart" - i.e. the items themselves - are marked as bought. The status passed in is set on the *order*.

Comment (melinath) [inline brambling/models.py]: Which is not the most intuitive, but also not something I want to refactor in this pull request.

**Pull request merged**